home/utils.py

```diff
@@ -204,37 +181,101 @@ def add_viber(row, page=None):
                 title=row["viber_title"],
                 enable_viberr=True,
                 viber_title=row["viber_title"],
-                viber_body=get_body(row, "viber_body", "viber_message"),
+                viber_body=get_body(viber_messages, "viber_message"),
                 viber_quick_replies=add_quick_replies(row["quick_replies"]),
                 locale=home_page.locale,
             )
         else:
             page.enable_viber = True
             page.viber_title = row["viber_title"]
-            page.viber_body = get_body(row, "viber_body", "viber_message")
+            page.viber_body = get_body(viber_messages, "viber_message")
             return page
 
+    def clean_row(row):
+        for field in (
+            "web_title",
+            "web_subtitle",
+            "web_body",
+            "whatsapp_title",
+            "whatsapp_body",
+            "next_prompt",
+            "viber_title",
+            "viber_body",
+            "messenger_title",
+            "messenger_body",
+            "parent",
+        ):
+            if field in row and row[field]:
+                row[field] = str(row[field]).strip()
+        return row
+
     if purge == "yes":
         ContentPage.objects.all().delete()
+        ContentPageIndex.objects.all().delete()
 
     if isinstance(locale, str):
         locale = Locale.objects.get(language_code=locale)
 
     home_page = HomePage.objects.get(locale__pk=locale.pk)
-    csv_file = file.read()
-    if isinstance(csv_file, bytes):
-        csv_file = csv_file.decode("utf-8")
-
-    reader = csv.DictReader(io.StringIO(csv_file))
-    for row in reader:
+    file = file.read()
+    lines = []
+    if filetype == "XLSX":
+        wb = load_workbook(filename=BytesIO(file))
+        ws = wb.active
+        ws.delete_cols(1, 7)
+        ws.delete_rows(1, 2)
```

Comment:
I'm wondering if we can name these columns and rows? I'm not sure what they represent, but something like `NOTES_COLUMN = 7`, `TRANSLATOR_COMMENTS_ROW = 2`, and then use those variables here instead

Reply:
Meg is actually working on making this dynamic in which case I don't think we will be able to name them. These rows represent the level the item is on. and with dynamic levels coming soon, we will have to remove an arbitrary amount of columns